### PR TITLE
fix(AppHeaderBar): pass onClick to the dropdown items

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
The onClick in dropdown item definition is not passed, so it's not called.

**What is the new behavior?**
It is passed .. and called !

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
could you please unsure that both onSelect + onClick are called with the new API ?

callback(event, {action}) ?

File: src/AppHeaderBar/AppHeaderBar.component.js [hunk 1]
```diff
@@ -26,16 +26,18 @@ renderNavItem.propTypes = {
 };
 
 export function renderDropdownItem(props, index) {
+	const { icon, name, ...rest } = props;
 	return (
-		<MenuItem key={index}>
-			<i className={props.icon} />
-			{props.name}
+		<MenuItem key={index} {...rest}>
+			<i className={icon} />
```
Comment (jmfrancois) [inline src/AppHeaderBar/AppHeaderBar.component.js]: will be fixed by using the new Icon ;)

Comment (jsomsanith) [inline src/AppHeaderBar/AppHeaderBar.component.js]: Done

**Comment by jsomsanith** [comment]
Discussed with @jmfrancois : enforced the callback arguments order (event, action) on onSelect

File: src/AppHeaderBar/AppHeaderBar.component.js [hunk 2]
```diff
@@ -26,21 +26,30 @@ renderNavItem.propTypes = {
 };
 
 export function renderDropdownItem(props, index) {
+	const { icon, name, ...rest } = props;
 	return (
-		<MenuItem key={index}>
-			<i className={props.icon} />
-			{props.name}
+		<MenuItem key={index} {...rest}>
+			<i aria-hidden="true" className={icon} />
```
Comment (frassinier) [inline src/AppHeaderBar/AppHeaderBar.component.js]: You should use the new `<Icon>` component :)

Comment (jmfrancois) [inline src/AppHeaderBar/AppHeaderBar.component.js]: agree

**Comment by jmfrancois** [comment]
please update the snapshot so we can merge this one

**Comment by jsomsanith** [comment]
Oh my bad. Sorry for that.
Done

**Pull request merged**